test_reporting/report_uploader.py

```diff
@@ -35,23 +35,29 @@ def _run_script():
     kusto_db = KustoConnector(args.db_name)
 
     if args.category == "test_result":
-        if args.json:
-            test_result_json = validate_junit_json_file(args.path_name)
-        else:
-            roots = validate_junit_xml_path(args.path_name)
-            test_result_json = parse_test_result(roots)
+        test_result_json = []
+        for path_name in args.path_list:
+            if args.json:
+                test_result_json.extend(validate_junit_json_file(path_name))
+            else:
+                roots = validate_junit_xml_path(path_name)
+                test_result_json.extend(parse_test_result(roots))
 
         tracking_id = args.external_id if args.external_id else ""
 
         kusto_db.upload_report(test_result_json, tracking_id)
```

Comment:
Is there a risk here that if one result upload fails, it results in a failure of all the result items?

Reply:
I think you also need to update the method signature here, to reflect new type (List).:
https://github.com/Azure/sonic-mgmt/blob/26285708eab7428c2b6250be03483d63f94cfb5c/test_reporting/report_data_storage.py#L105

I am also not sure how the upload worked, as `test_result_json` here is a list, which is accessed as a dict later in the upload process:

 https://github.com/Azure/sonic-mgmt/blob/26285708eab7428c2b6250be03483d63f94cfb5c/test_reporting/report_data_storage.py#L150

Reply:
@daall  please chime in here. I don't think I changed data type. But report_json is claimed to be Dict. Can you confirm?

Reply:
@vaibhavhd I don't think we have real risk here in terms of group operations. Our test results currently is a single file. We will rerun the task if upload fails. It is better to fail all data instead of uploading a partial data. IMO.

Reply:
I don't think you changed the data type, we're calling the same method once for each path that's passed in, right?

Reply:
Oh, no, I think @vaibhavhd is right. `report_json` should be a dict, not a list.

Reply:
ok, I see it as well now. Sorry for the confusion.